### ISSUE TITLE
Parsing error when attempting to add a second link on a line

### ISSUE THREAD

**Issue by vimalloc** [epwalsh/obsidian.nvim#212]
### 🐛 Describe the bug

Thank you for this wonderful extension!

I'm running into a parsing error when I try to add a second link to a line that already has a link, and I am struggling to figure out why that is happening. Here is a screen capture showing how the error is triggered:

https://github.com/epwalsh/obsidian.nvim/assets/238933/0a2a434c-9b62-4951-af9d-8386ea22c8c2

The parsing error comes up when i type in a single `[` character (although you cannot see it on the screen capture, because the error message writes over it). But if I type in the second `[` character and start typing in a link, the auto-fill still works, just with that error message being displayed on the page.

I would be 0% surprised if this was a configuration problem or something stupid I have done on my end. If you have any ideas on what might be causing this or how to troubleshoot it further, I would really appreciate it! Here are the configs I have setup around `nvim-cmp` and `obsidian.nvim`, 

### obsidian.nvim config (using lazy.vim)
```
local obsidian      = { "epwalsh/obsidian.nvim",
                        lazy = true,
                        event = { "BufReadPre " .. vim.fn.expand "~" .. "/Documents/Dnd_Vault/**.md",
                                  "BufNewFile " .. vim.fn.expand "~" .. "/Documents/Dnd_Vault/**.md" },
                        opts = {
                          workspaces = { { name = "dnd", path = "~/Documents/Dnd_Vault" } },
                          completion = { nvim_cmp = true, min_chars = 2 },
                          disable_frontmatter = true,
                        },
                        dependencies = { "nvim-lua/plenary.nvim", "hrsh7th/nvim-cmp" } }
```

### nvim-cmp config
```
local cmp = require'cmp'

cmp.setup({
  snippet = {
    expand = function(args)
      require('luasnip').lsp_expand(args.body)
    end,
  },
  window = {
    completion = cmp.config.window.bordered(),
    -- documentation = cmp.config.window.bordered(),
  },
  mapping = {
    ["<C-p>"] = cmp.mapping.select_prev_item(),
    ["<C-n>"] = cmp.mapping.select_next_item(),
    ["<S-Tab>"] = cmp.mapping.select_prev_item(),
    ["<Tab>"] = cmp.mapping.select_next_item(),
    ["<C-d>"] = cmp.mapping.scroll_docs(-4),
    ["<C-f>"] = cmp.mapping.scroll_docs(4),
    ["<C-Space>"] = cmp.mapping(cmp.mapping.complete(), { "i", "c" }),
    ["<C-e>"] = cmp.mapping.close(),
    ["<CR>"] = cmp.mapping.confirm {
      behavior = cmp.ConfirmBehavior.Insert,
    },
  },
  sorting = {
    priority_weight = 2,
    comparators = {
      require("copilot_cmp.comparators").prioritize,

      -- Below is the default comparitor list and order for nvim-cmp
      cmp.config.compare.offset,
      -- cmp.config.compare.scopes, --this is commented in nvim-cmp too
      cmp.config.compare.exact,
      cmp.config.compare.score,
      cmp.config.compare.recently_used,
      cmp.config.compare.locality,
      cmp.config.compare.kind,
      cmp.config.compare.sort_text,
      cmp.config.compare.length,
      cmp.config.compare.order,
    },
  },
  sources = cmp.config.sources({
    { name = 'nvim_lsp' },
    { name = 'luasnip' },
    { name = "copilot", group_index = 2 },
  }, {
    { name = 'buffer' },
  })
})

-- Use buffer source for `/` (if you enabled `native_menu`, this won't work anymore).
cmp.setup.cmdline('/', {
  mapping = cmp.mapping.preset.cmdline(),
  sources = {
    { name = 'buffer' }
  }
})

-- Use cmdline & path source for ':' (if you enabled `native_menu`, this won't work anymore).
cmp.setup.cmdline(':', {
  mapping = cmp.mapping.preset.cmdline(),
  sources = cmp.config.sources({
    { name = 'path' }
  }, {
    { name = 'cmdline' }
  })
})
```
And for good measure, checkhealth and lsp information, to see if those were causing any problems

### :ObsidianCheckHealth
```
[Obsidian] All good!
Vault configured to '/Users/lilygilbert/Documents/Dnd_Vault'
```

### :LspInfo
```
0 client(s) attached to this buffer:
```

### Versions

```
$ nvim --version
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1696795921

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.9.4/share/nvim"

Run :checkhealth for more info

$ nvim --headless -c 'lua print("Obsidian v" .. require("obsidian").VERSION)' -c q
Obsidian v1.15.0
```

**Comment by epwalsh** [comment]
Hey @vimalloc I'm able to reproduce this in my setup too. I'm looking into it.

**Comment by epwalsh** [comment]
Turns out there was a simple fix! #213

**Comment by vimalloc** [comment]
Incredible! Thanks for the quick fix! <3
